runtime/moonbase/src/precompiles.rs

```diff
@@ -98,6 +102,10 @@ where
 			a if a == hash(2051) => {
 				Some(DemocracyWrapper::<R>::execute(input, target_gas, context))
 			}
+			// If the address matches asset prefix, the we route through the asset precompile set
+			a if &a.to_fixed_bytes()[0..4] == ASSET_PRECOMPILE_ADDRESS_PREFIX => {
```

Comment:
Consider adding a helper method on your precompile set called eg is_matching_prefix. Then there is no chance of accidentally using two different values

Reply:
So the problem is the following. I want to check if the prefix matches at this point, as in the future we could have other precompiles in a different prefix and might need to route through them as well. And I would not like to implement a helper identical for all runtimes, as the assumption is that the accountIds might be different across runtimes (that's why they are generic). I could implement a trait but then I would require the implementation of such a trait in the runtime/precompileset for this function, which is something I dont like much either. So instead for now I am going to make that constant public and I am going to use it in our runtime to implement the AccountIdAssetIdConversion traitm and that way we ensure there isnt a missmatch between the value we use. We can change this in the future, but for now I think this is the cleaner way

Reply:
Fine with me. :+1: